Lib/test/test_frame.py

```diff
@@ -442,5 +461,28 @@ def dummy():
         # The following line should not cause a segmentation fault.
         self.assertIsNone(frame.f_back)
 
+class LocalsTest(unittest.TestCase):
+    """
+    Tests for locals.
+    """
+
+    def test_locals_cur_frame_issue113939(self):
+        class C:
+            pass
+        wr = None
+        def inner():
+            nonlocal wr
+            c = C()
+            wr = weakref.ref(c)
+            1/0
+        try:
+            inner()
+        except ZeroDivisionError as exc:
+            support.gc_collect()
+            self.assertIsNotNone(wr())
+            print(exc.__traceback__.tb_frame.f_locals)
```

Comment:
print?

Reply:
I actually just want to access `exc.__traceback__.tb_frame.f_locals` here (trigger a `getattr`), which is important for the test. I just thought that also printing the result might be interesting for debugging purpose, but we could also leave away the print.